x-pack/plugins/telemetry_collection_xpack/server/telemetry_collection/get_stats_with_xpack.ts

```diff
@@ -21,14 +21,23 @@ export const getStatsWithXpack: StatsGetter<{}, TelemetryAggregatedStats> = asyn
   const clustersLocalStats = await getLocalStats(clustersDetails, config, context);
   const xpack = await getXPackUsage(esClient).catch(() => undefined); // We want to still report something (and do not lose the license) even when this method fails.
 
-  return clustersLocalStats.map((localStats) => {
-    if (xpack) {
-      return {
-        ...localStats,
-        stack_stats: { ...localStats.stack_stats, xpack },
-      };
-    }
+  return clustersLocalStats
+    .map((localStats) => {
+      if (xpack) {
+        return {
+          ...localStats,
+          stack_stats: { ...localStats.stack_stats, xpack },
+        };
+      }
 
-    return localStats;
-  });
+      return localStats;
+    })
+    .reduce((acc, stats) => {
+      // Concatenate the telemetry reported via monitoring as additional payloads instead of reporting it inside of stack_stats.kibana.plugins.monitoringTelemetry
+      const monitoringTelemetry = stats.stack_stats.kibana?.plugins?.monitoringTelemetry;
+      if (monitoringTelemetry) {
+        delete stats.stack_stats.kibana!.plugins.monitoringTelemetry;
+      }
+      return [...acc, stats, ...(monitoringTelemetry || [])];
+    }, [] as TelemetryAggregatedStats[]);
```

Comment:
It may feel like we are mixing concepts, and that this telemetry collection strategy needs to know about other existing collectors to reshape the format that is ultimately sent to the telemetry service. But, afaik, that's one of the duties in the telemetry collection strategies (the OSS strategy also extracts some collectors to shape them into something else).

On an unrelated note, I'm planning on a follow-up PR that will change the behaviour of the telemetryCollectionManager: It will use this X-Pack strategy as a replacement of the OSS strategy (instead of the current priority-based loop, trying the next lower-priority strategy if the previous one fails).